### PR TITLE
adding advertised listeners property, additional hostnames in auto generated certs, check if keystores/truststores exists before regen

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

Can now have a hostname on each listener (which helps in aws land) and those hostnames get set in the autogenerated certs in the extension

Also now ssl role has a task that checks if keystores/truststores are created before running everything

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: roles/confluent.kafka_broker/tasks/main.yml [hunk 1]
```diff
@@ -29,6 +29,11 @@
     system: yes
     group: "{{kafka_broker.group}}"
 
+# prestep to ssl role, need to gather all hostnames (ie external/internal for aws) and pass to ssl role by some var
+
+# - set_fact:
```
Comment (domenicbove) [inline roles/confluent.kafka_broker/tasks/main.yml]: remove commented code

File: tasks/certificate_authority.yml [hunk 2]
```diff
@@ -14,6 +14,13 @@
       -subj '/CN=ca1.test.confluent.io/OU=TEST/O=CONFLUENT/L=PaloAlto/S=Ca/C=US' \
       -passin pass:{{ssl_self_signed_ca_password}} -passout pass:{{ssl_self_signed_ca_password}}
 
+# TODO should create a truststore for testing purposes
+# - name: Create Truststore and Import the CA Cert
+#   shell: |
+#     keytool -noprompt -keystore {{truststore_path}} -storetype pkcs12 \
```
Comment (domenicbove) [inline tasks/certificate_authority.yml]: remove commented code

File: roles/confluent.kafka_broker/tasks/main.yml [hunk 3]
```diff
@@ -75,7 +76,7 @@
     group: "{{kafka_broker.group}}"
     owner: "{{kafka_broker.user}}"
     mode: '764'
-    recurse: yes
+    # recurse: yes
```
Comment (domenicbove) [inline roles/confluent.kafka_broker/tasks/main.yml]: put in separate PR

**Comment by JumaX** [comment]
@domenicbove looks like there is a conflict, possibly due to a change I merged earlier relating to the variable name for self signed certs.  Can you review?
